hazelcast/src/main/java/com/hazelcast/jet/impl/JobCoordinationService.java

```diff
@@ -850,7 +850,13 @@ private JobAndSqlSummary getJobAndSqlSummary(LightMasterContext lmc) {
      */
     public CompletableFuture<JobConfig> updateJobConfig(long jobId, @Nonnull DeltaJobConfig deltaConfig) {
         return callWithJob(jobId,
-                masterContext -> masterContext.updateJobConfig(deltaConfig),
+                masterContext -> {
+                    if (!Util.isJobSuspendable(masterContext.jobConfig())) {
+                        throw new IllegalStateException("The job '" + masterContext.jobConfig().getName()
+                                + "' is not suspendable, can't perform `updateJobConfig()`");
```

Comment:
common logic/validation should be in the implementation not invocation. Move it to MasterContext.updateJobConfig.